src/main/java/nb/kafka/operator/HealthServer.java

```diff
@@ -0,0 +1,43 @@
+package nb.kafka.operator;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.undertow.Undertow;
+import io.undertow.server.handlers.PathHandler;
+import io.undertow.util.StatusCodes;
+
+public class HealthServer {
+
+  private static final Logger logger = LoggerFactory.getLogger(HealthServer.class);
+  private static final int DEFAULT_PORT = 8080;
+  private static final String DEFAULT_HOST = "0.0.0.0";
+
+  public static void start(AppConfig config) {
+    PathHandler pathHandler = new PathHandler();
+    pathHandler.addPrefixPath("/healthy", exchange -> exchange.setStatusCode(StatusCodes.OK));
+    pathHandler.addPrefixPath("/ready", exchange -> {
+      if (isKafkaAvailable(config))
+        exchange.setStatusCode(StatusCodes.OK);
+      else
+        exchange.setStatusCode(StatusCodes.INTERNAL_SERVER_ERROR);
+    });
+    logger.info("Starting health checks server on port: " + DEFAULT_PORT);
+    Undertow server = Undertow.builder().addHttpListener(DEFAULT_PORT, DEFAULT_HOST)
```

Comment:
Do we need undertow here? Can we use something simpler like the one already coming from Prometheus exporter (HTTPServer)?

Reply:
The problem with io.prometheus.client.exporter.HTTPServer is that it is not a general purpose web server, we can only give it the prometheus registry and there is no method to create custom endpoints on it.

Reply:
As discussed offline, will be using com.sun.net.httpserver.HttpServer

Reply:
Hi @nbogojevic , I have used undertow as a lightweight http server. I got your point to reduce the number of dependencies. Implement health server using prometheus will bring less code readability and that would mix up the different usage of endpoints in the same code base.
In our last [commit](https://github.com/benjaminbillet/kafka-operator/commit/add99727093cc17fcaad71e3fafb1fd3b13b2209#diff-4421998a9e51f7f27d441bc5c712503fR14), we have implemented the HealthCheckServer using  `com.sun.net.httpserver.HttpServer`.
We have removed undertow dependency.